docs/developer/decision-records/2022-04-21-dpf-blob-data-transfer/README.md

```diff
@@ -0,0 +1,38 @@
+# Blob storage transfer
+
+ADR describing the blob storage transfer end to end flow between 2 participants.
+
+## Description
+
+The data-plane-azure-storage extension can be used on DPF to support blob transfers.
+A client can trigger a blob transfer on the consumer side via the Data Management API.
+The client needs to use the managedResources=true option in its HTTP request, otherwise the transfer will not work. Client would not be aware that an error occurred, it would just never see the transferProcess with completed state when polling for the result.   
+Storage accounts access key should be stored in advanced in Keyvaults. The consumer can generate SAS token to give the provider the possibility to write data to its container.
+
+## Sequence diagram
+
+The following sequence diagram describes the flow to transfer a blob from a Provider storage account to a consumer storage account.
+It starts from the client triggering the transfer on the consumer side and finishes when the consumer deletes the blob
+after the client triggered the data deletion.
+
+![blob-transfer](../../../diagrams/blob-transfer.png)
+
+1. The client calls the data management API to trigger a transfer process. managedResources needs to be set to true, to make sure that the consumer provisions the blob container. If it is not set to true, the transfer will not work. Then, the transfer would never be completed. 
+2. Consumer gets the destination storage account access key in its Vault.  
+3. Consumer creates a container where the Provider DPF may write blobs. The container is created only if the client specifies managedResources=true.
+   The [ObjectStorageProvisioner](../../../../extensions/azure/blobstorage/blob-provision/src/main/java/org/eclipse/dataspaceconnector/provision/azure/blob/ObjectStorageProvisioner.java) is responsible for provisioning the container and for generating a SAS token to access the container.
+   It creates the container and the SAS token by using the [BlobStoreApi](../../../../extensions/azure/blobstorage/blob-core/src/main/java/org/eclipse/dataspaceconnector/azure/blob/core/api/BlobStoreApi.java). [BlobStoreApiImpl](../../../../extensions/azure/blobstorage/blob-core/src/main/java/org/eclipse/dataspaceconnector/azure/blob/core/api/BlobStoreApiImpl.java) retrieves the storage account access key in the consumer vault. Then it can create the container and generate the SAS token.  
```

Comment:
nit: If you agree then maybe we can remove this part as it is not very clear.
```suggestion
```

Reply:
Well I think we should document that generating a SAS token requires a storage account key, and the latter is retrieved from the key vault. Perhaps we could link to Azure doc for this

Reply:
@cpeeyush I removed the details about BlobStoreApiImpl class but I specified that the storage account key needs to be present in the Vault.